QPM/SymLinker.Linker/Linker.cs

```diff
@@ -0,0 +1,269 @@
+﻿using SymLinker.Linker.LinkCreators;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+
+// Stolen from https://github.com/huffSamuel/SymLinker
+// Cleaned up by Fern, because the old code was horrible
+namespace SymLinker.Linker
+{
+    public class Linker
+    {
+        #region Actions
+        // Attach to these to get warnings, errors, and informaion out of SymLinkerCore
+        public Action<string> OnWarn;
```

Comment:
These should be `event`s instead of delegate fields, ensures proper usage of them and no forced assignment.